flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/runtime/aggregate/MaxAggregate.scala

```diff
@@ -48,8 +48,16 @@ abstract class MaxAggregate[T](implicit ord: Ordering[T]) extends Aggregate[T] {
   override def merge(intermediate: Row, buffer: Row): Unit = {
     val partialValue = intermediate.productElement(maxIndex).asInstanceOf[T]
     val bufferValue = buffer.productElement(maxIndex).asInstanceOf[T]
-    val max: T = if (ord.compare(partialValue, bufferValue) > 0) partialValue else bufferValue
-    buffer.setField(maxIndex, max)
+    if (partialValue == null && bufferValue == null) {
```

Comment:
Can we reorder the `if else` cases to move the common case of `partialValue != null && bufferValue != null` to the top?